From 8b8e86b2ec6420f4de463e8e2905191f675649ef Mon Sep 17 00:00:00 2001
From: KancyJoe <kancy2333@outlook.com>
Date: Sat, 15 Nov 2025 12:34:37 +0000
Subject: [PATCH 2/7] ignore basetool warnings

---
 BaseTools/Source/C/Makefiles/header.makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/BaseTools/Source/C/Makefiles/header.makefile b/BaseTools/Source/C/Makefiles/header.makefile
index 40604b37..6a38a251 100644
--- a/BaseTools/Source/C/Makefiles/header.makefile
+++ b/BaseTools/Source/C/Makefiles/header.makefile
@@ -70,9 +70,13 @@ BUILD_OPTFLAGS = -O2 $(EXTRA_OPTFLAGS)
 ifeq ($(DARWIN),Darwin)
 # assume clang or clang compatible flags on OS X
 BUILD_CFLAGS = -MD -fshort-wchar -fno-strict-aliasing -Wall -Werror \
+-Wno-stringop-truncation -Wno-vla-parameter -Wno-stringop-overflow \
+-Wno-use-after-free -Wno-dangling-pointer \
 -Wno-deprecated-declarations -Wno-self-assign -Wno-unused-result -nostdlib -g
 else
 BUILD_CFLAGS = -MD -fshort-wchar -fno-strict-aliasing -fwrapv \
+-Wno-stringop-truncation -Wno-vla-parameter -Wno-stringop-overflow \
+-Wno-use-after-free -Wno-dangling-pointer \
 -fno-delete-null-pointer-checks -Wall -Werror \
 -Wno-deprecated-declarations \
 -Wno-unused-result -nostdlib -g
-- 
2.43.0

