From bfca9f6515671a47daad0826b7c6154c2b0dab06 Mon Sep 17 00:00:00 2001
From: KancyJoe <kancy2333@outlook.com>
Date: Thu, 20 Nov 2025 12:52:44 +0000
Subject: [PATCH 06/12] avb: disable unlock warning

---
 QcomModulePkg/Library/avb/VerifiedBoot.c | 22 +++++++++++-----------
 1 file changed, 11 insertions(+), 11 deletions(-)

diff --git a/QcomModulePkg/Library/avb/VerifiedBoot.c b/QcomModulePkg/Library/avb/VerifiedBoot.c
index bbeb7c1d..9706c108 100644
--- a/QcomModulePkg/Library/avb/VerifiedBoot.c
+++ b/QcomModulePkg/Library/avb/VerifiedBoot.c
@@ -1844,17 +1844,17 @@ DisplayVerifiedBootScreen (BootInfo *Info)
     }
     break;
   case ORANGE:
-    if (FfbmStr[0] != '\0' && !TargetBuildVariantUser ()) {
-      DEBUG ((EFI_D_VERBOSE, "Device will boot into FFBM mode\n"));
-    } else {
-      Status = DisplayVerifiedBootMenu (DISPLAY_MENU_ORANGE);
-      if (Status == EFI_SUCCESS) {
-        WaitForExitKeysDetection ();
-      } else {
-        DEBUG (
-            (EFI_D_INFO, "Device is unlocked, Skipping boot verification\n"));
-      }
-    }
+//    if (FfbmStr[0] != '\0' && !TargetBuildVariantUser ()) {
+//      DEBUG ((EFI_D_VERBOSE, "Device will boot into FFBM mode\n"));
+//    } else {
+//      Status = DisplayVerifiedBootMenu (DISPLAY_MENU_ORANGE);
+//      if (Status == EFI_SUCCESS) {
+//        WaitForExitKeysDetection ();
+//      } else {
+//        DEBUG (
+//            (EFI_D_INFO, "Device is unlocked, Skipping boot verification\n"));
+//      }
+//    }
     break;
   default:
     break;
-- 
2.43.0

