From d7c1fa03696c74e28f2aaca10cf28339cabe92fc Mon Sep 17 00:00:00 2001
From: KancyJoe <kancy2333@outlook.com>
Date: Thu, 4 Dec 2025 10:37:12 +0000
Subject: [PATCH 01/13] disable basetool clean

---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index 77a1cd068d..93ba168f5f 100644
--- a/makefile
+++ b/makefile
@@ -373,7 +373,7 @@ ABL_FV_IMG: $(EDK_TOOLS_PATH_MARK_FILE)
 	cp $(BUILD_ROOT)/FV/FVMAIN_COMPACT.Fv $(ABL_FV_IMG)
 
 $(EDK_TOOLS_GENERATE_CLEAN): $(EDK_TOOLS_PATH_MARK_FILE)
-	@$(MAKEPATH)make -C $(BUILDDIR)/BaseTools/Source/C clean > /dev/null
+#	@$(MAKEPATH)make -C $(BUILDDIR)/BaseTools/Source/C clean > /dev/null
 	touch $(EDK_TOOLS_GENERATE_CLEAN)
 
 BASETOOLS_CLEAN: ABL_FV_IMG
-- 
2.43.0

