From 6039c452bd6525ceef2f0cb439a3c19e0e4bac6a Mon Sep 17 00:00:00 2001
From: KancyJoe <kancy2333@outlook.com>
Date: Thu, 4 Dec 2025 10:51:45 +0000
Subject: [PATCH 06/13] avb: disable unlock warning

---
 QcomModulePkg/Library/avb/VerifiedBoot.c | 42 ++++++++++++------------
 1 file changed, 21 insertions(+), 21 deletions(-)

diff --git a/QcomModulePkg/Library/avb/VerifiedBoot.c b/QcomModulePkg/Library/avb/VerifiedBoot.c
index 2e8a0d95a5..ed4baffa7b 100644
--- a/QcomModulePkg/Library/avb/VerifiedBoot.c
+++ b/QcomModulePkg/Library/avb/VerifiedBoot.c
@@ -1845,33 +1845,33 @@ DisplayVerifiedBootScreen (BootInfo *Info)
     }
     break;
   case ORANGE:
-    if (FfbmStr[0] != '\0' && !TargetBuildVariantUser ()) {
-      DEBUG ((EFI_D_VERBOSE, "Device will boot into FFBM mode\n"));
-    } else {
-      Status = DisplayVerifiedBootMenu (DISPLAY_MENU_ORANGE);
-      if (Status == EFI_SUCCESS) {
-        WaitForExitKeysDetection ();
-      } else {
-        DEBUG (
-            (EFI_D_INFO, "Device is unlocked, Skipping boot verification\n"));
-      }
-    }
+//    if (FfbmStr[0] != '\0' && !TargetBuildVariantUser ()) {
+//      DEBUG ((EFI_D_VERBOSE, "Device will boot into FFBM mode\n"));
+//    } else {
+//      Status = DisplayVerifiedBootMenu (DISPLAY_MENU_ORANGE);
+//      if (Status == EFI_SUCCESS) {
+//        WaitForExitKeysDetection ();
+//      } else {
+//        DEBUG (
+//            (EFI_D_INFO, "Device is unlocked, Skipping boot verification\n"));
+//      }
+//    }
     break;
   default:
     break;
   }
 
   /* dm-verity warning */
-  if ( !IsEnforcing () &&
-       !Info->BootIntoRecovery) {
-      Status = DisplayVerifiedBootMenu (DISPLAY_MENU_EIO);
-      if (Status == EFI_SUCCESS) {
-        WaitForExitKeysDetection ();
-      } else {
-        DEBUG ((EFI_D_INFO, "The dm-verity is not started in restart mode." \
-              "\nWait for 30 seconds before proceeding\n"));
-      }
-  }
+//  if ( !IsEnforcing () &&
+//       !Info->BootIntoRecovery) {
+//      Status = DisplayVerifiedBootMenu (DISPLAY_MENU_EIO);
+//      if (Status == EFI_SUCCESS) {
+//        WaitForExitKeysDetection ();
+//      } else {
+//        DEBUG ((EFI_D_INFO, "The dm-verity is not started in restart mode." \
+//              "\nWait for 30 seconds before proceeding\n"));
+//      }
+//  }
 
   return EFI_SUCCESS;
 }
-- 
2.43.0

