From 45a926ef859f2dde2b80c1519dc5ec6126d7269e Mon Sep 17 00:00:00 2001
From: KancyJoe <kancy2333@outlook.com>
Date: Thu, 4 Dec 2025 10:55:22 +0000
Subject: [PATCH 07/13] only load fv drivers when loading fastboot

---
 QcomModulePkg/Application/LinuxLoader/LinuxLoader.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/QcomModulePkg/Application/LinuxLoader/LinuxLoader.c b/QcomModulePkg/Application/LinuxLoader/LinuxLoader.c
index a445eabdf7..7daf123a7f 100644
--- a/QcomModulePkg/Application/LinuxLoader/LinuxLoader.c
+++ b/QcomModulePkg/Application/LinuxLoader/LinuxLoader.c
@@ -329,9 +329,6 @@ LinuxLoaderEntry (IN EFI_HANDLE ImageHandle, IN EFI_SYSTEM_TABLE *SystemTable)
 
   BootStatsSetTimeStamp (BS_BL_START);
 
-  /* Load Drivers from current FV */
-  LoadDriversFromCurrentFv(ImageHandle);
-
   /* check if it is NetworkBoot, FlashlessBoot or Fastboot */
   if (IsMultiBoot ()) {
     Val = GetBootDeviceType ();
@@ -504,6 +501,9 @@ flashless_boot:
   }
 
 fastboot:
+  /* Load Drivers from current FV */
+  LoadDriversFromCurrentFv(ImageHandle);
+
   if (FlashlessBoot ||
       NetworkBoot) {
     DEBUG ((EFI_D_ERROR, "No fastboot support for flashless chipsets,"
-- 
2.43.0

